js/src/vm/PortableBaselineInterpret.cpp

```diff
@@ -1123,6 +1123,84 @@ ICInterpretOps(BaselineFrame* frame, VMFrameManager& frameMgr, State& state,
     DISPATCH_CACHEOP();
   }
 
+  CACHEOP_CASE(GuardDynamicSlotIsNotObject) {
+    ObjOperandId objId = icregs.cacheIRReader.objOperandId();
+    uint32_t slotOffset = icregs.cacheIRReader.stubOffset();
+    JSObject* obj = reinterpret_cast<JSObject*>(icregs.icVals[objId.id()]);
+    uint32_t slot = cstub->stubInfo()->getStubRawInt32(cstub, slotOffset);
+    NativeObject* nobj = &obj->as<NativeObject>();
+    HeapSlot* slots = nobj->getSlotsUnchecked();
+    Value actual = slots[slot / sizeof(Value)];
```

Comment:
Just to double-check (I think you mentioned this but...) -- this `slot / sizeof(Value)` has been empirically verified correct w.r.t. the below? If we haven't already, it might be good to do the usual trick of switching it (to `slots[slot]`) and seeing if tests break; and then leaving a comment here just like you did below to note the difference on both sides.

Reply:
Good catch! No, the test suite apparently can't distinguish between these two implementations, but given the choice of field names you're probably right. I've pushed the probable-fix and corresponding comment.

Reply:
Ah, interesting. I search for uses of this CacheIR op and at least [this one](https://searchfox.org/mozilla-central/rev/3fe52b82b0a00829d0a8224b4bd767cb20b7a27d/js/src/jit/CacheIR.cpp#11259) seems to indeed imply that it's a slot number rather than byte offset.